kernel/queue.c

```diff
@@ -303,7 +303,7 @@ static void *k_queue_poll(struct k_queue *queue, s32_t timeout)
 		val = z_queue_node_peek(sys_sflist_get(&queue->data_q), true);
 		irq_unlock(key);
 
-		if (!val && timeout != K_FOREVER) {
+		if ((val == NULL) && timeout != K_FOREVER) {
```

Comment:
This isn't just superfluous, but inconsistent parens. We can't really use parens on left side of && and not use on right. (It should be consistent - and none required left or right).

Reply:
ok, I can do that. In this case I have created a script to do these changes.
I can manually change this occurrence or change the script to enforce parentheses in the other expressions.

For now I'll just change this expression as you asked.

Reply:
@pfalcon got confuse now. Most zephyr's code is doing

```
if (x && (y != K_SOMETHING)) { ... }
```

This case is not the most common, it's easier changing adding the parentheses int the second expression. One thing that MISRA-C tries to do is decrease the chance of a developer introduce errors, in this case the parentheses is something valid IMHO because there are less chance of a programmer introduce an error caused by precedence of operators.


That's my opinion, I'm totally open to remove the parentheses in all cases, I just prefer send the fix prior bring a lot of discussion that normally takes infinity time. We need to move forward in this MISRA task :)

Reply:
> Most zephyr's code is doing
> if (x && (y != K_SOMETHING)) { ... }

I did `grep " && " *` in kernel/ , and word "most" is not which I'd use to describe the situation. Indeed, both extra-parens and no-parens styles are used throughout. My concern here specifically was inconsistent use of parens one side vs another side of operator, but as we can see, the problem is deeper than that, definitely not to address this time. Let me dismiss my -1 so just current task at hand can proceed. Thanks for paying attention to matters which arise around addressing the MISRA matters.